assets/react/controllers/Incident/incidentStore.js

```diff
@@ -0,0 +1,59 @@
+import React from "react";
+import { Provider } from "react-redux";
+import { createSlice, configureStore } from "@reduxjs/toolkit";
+
+const initialState = {
+  loaded: false,
+  incident: null,
+  order: null,
+  images: [],
+  transporterEnabled: false,
+};
+
+const incidentSlice = createSlice({
+  name: "incident",
+  initialState,
+  reducers: {
+    setLoaded(state, action) {
+      state.loaded = action.payload;
+    },
+    setIncident(state, action) {
+      state.incident = action.payload;
+    },
+    setOrder(state, action) {
+      state.order = action.payload;
+    },
+    setImages(state, action) {
+      state.images = action.payload;
+    },
+    setEvents(state, action) {
+      state.incident.events = action.payload;
+    },
+    setTransporterEnabled(state, action) {
+      state.transporterEnabled = action.payload;
+    },
+  },
+});
+
+const store = configureStore({
+  reducer: incidentSlice.reducer,
+});
+
+export const {
+  setLoaded,
+  setIncident,
+  setOrder,
+  setImages,
+  setEvents,
+  setTransporterEnabled,
+} = incidentSlice.actions;
+
+export default store;
+
+export const useStore =
```

Comment:
That's a nice trick. Is it to access redux store from the components added with `symfony/ux-react`?

Reply:
Maybe `useStore` is not a good name, though, because it's not a hook https://react.dev/reference/rules/rules-of-hooks#only-call-hooks-at-the-top-level

Reply:
You're correct, and the name is even more inaccurate with 'store.' You can access the store directly without wrapping the component with that function. It simply enables store reactivity.

We could use:
- connectWithRedux
- reduxWrapper
- reactiveReduxWrapper